### PR TITLE
Salesforce timer-based event sources

### PR DESCRIPTION
**Pull request by jverce**
## Description
This PR provides an alternative to the previous Salesforce event sources, and does not require Apex code deployment (as described in #626)

## Changelog
* Refactor Salesforce event sources for every SObject types to be
  timer-based
* Remove everything related to the SOAP API and Apex, and rely solely
  on the REST API `getUpdated` and `getDeleted` endpoints

### CODE REVIEW

File: components/salesforce/common.js [hunk 1]
```diff
@@ -1,152 +1,77 @@
...
-      return this.standardizeName("webhook");
-    },
-    _isValidSource(event) {
-      const {
-        "x-webhook-token": webhookToken,
-      } = event.headers;
-      const secretToken = this.db.get("secretToken");
-      return webhookToken === secretToken;
+    isValidSObject(sobject) {
+      return sobject.replicateable;
```
Comment (dylburger) [inline components/salesforce/common.js:53]: Could you describe what the function of the `replicateable` field is briefly in a comment in this method?

File: components/salesforce/sources/new-object/new-object.js [hunk 2]
```diff
@@ -2,26 +2,31 @@ const startCase = require("lodash/startCase");
 
 const common = require("../../common");
 
-const EVENT_SOURCE_NAME = "New Object (Instant)";
-
 module.exports = {
   ...common,
-  name: EVENT_SOURCE_NAME,
+  name: "New Object",
```
Comment (dylburger) [inline components/salesforce/sources/new-object/new-object.js:7]: Is this the name Zapier assigns? Maybe we can edit the description to make the fact that you choose the object type clear

File: components/salesforce/sources/new-object/new-object.js [hunk 3]
```diff
@@ -2,26 +2,31 @@ const startCase = require("lodash/startCase");
...
   key: "salesforce-new-object",
   description: "Emit an event when an object is created",
```
Comment (dylburger) [inline components/salesforce/sources/new-object/new-object.js:9]: Re: the above comment, maybe add something here re: the fact that this source lets you listen for new objects of any type, and that you choose the type when creating the source.

File: components/salesforce/sources/new-object/new-object.js [hunk 4]
```diff
@@ -30,31 +35,33 @@ module.exports = {
...
+        startTimestamp,
+        endTimestamp,
+      } = eventData;
+      const {
+        ids,
+        latestDateCovered,
+      } = await this.salesforce.getUpdatedForObjectType(this.objectType, startTimestamp, endTimestamp);
+
+      // By the time we try to retrieve an item, it might've been deleted. This
+      // will cause `getSObject` to throw a 404 exception, which will reject it's
```
Comment (dylburger) [inline components/salesforce/sources/new-object/new-object.js:49]: "reject its promise"

**Review by dylburger — APPROVED** [review]

File: components/salesforce/common.js (hunk 1 above)
Comment (jverce) [reply components/salesforce/common.js:53]: Will do

File: components/salesforce/sources/new-object/new-object.js (hunk 2 above)
Comment (jverce) [reply components/salesforce/sources/new-object/new-object.js:7]: They actually use `New Record in Salesforce`. We could rename it to that?

File: components/salesforce/sources/new-object/new-object.js (hunk 4 above)
Comment (jverce) [reply components/salesforce/sources/new-object/new-object.js:49]: Oops, fixed.

File: components/salesforce/sources/new-object/new-object.js (hunk 3 above)
Comment (jverce) [reply components/salesforce/sources/new-object/new-object.js:9]: Will do. How long are these descriptions supposed to be? Do we have a limit on their length?

Comment (jverce) [reply components/salesforce/sources/new-object/new-object.js:9]: OK, I tried with a multiline string and it's handled fine by the UI so I'll go with that!

**Pull request merged**